src/firmware/boards/robot_stm32h7/freertos.c

```diff
@@ -31,22 +31,65 @@
+#include "firmware/app/primitives/primitive_manager.h"
+#include "firmware/app/world/firmware_robot.h"
+#include "firmware/app/world/firmware_world.h"
+#include "firmware/boards/robot_stm32h7/io/charger.h"
+#include "firmware/boards/robot_stm32h7/io/chicker.h"
+#include "firmware/boards/robot_stm32h7/io/dribbler.h"
 #include "firmware/boards/robot_stm32h7/io/drivetrain.h"
 #include "firmware/boards/robot_stm32h7/io/network_logger.h"
 #include "firmware/boards/robot_stm32h7/io/power_monitor.h"
+#include "firmware/boards/robot_stm32h7/io/primitive_executor.h"
+#include "firmware/boards/robot_stm32h7/io/proto_multicast.h"
 #include "firmware/boards/robot_stm32h7/io/proto_multicast_communication_profile.h"
-#include "firmware/boards/robot_stm32h7/io/proto_multicast_communication_tasks.h"
+#include "firmware/boards/robot_stm32h7/io/robot_status.h"
+#include "firmware/boards/robot_stm32h7/io/uart_logger.h"
 #include "firmware/boards/robot_stm32h7/io/ublox_odinw262_communicator.h"
+#include "firmware/boards/robot_stm32h7/io/vision.h"
 #include "firmware/boards/robot_stm32h7/tim.h"
 #include "firmware/boards/robot_stm32h7/usart.h"
-#include "shared/constants.h"
+#include "firmware/shared/physics.h"
 #include "shared/proto/robot_log_msg.nanopb.h"
 #include "shared/proto/robot_status_msg.nanopb.h"
 #include "shared/proto/tbots_software_msgs.nanopb.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
+typedef StaticTask_t osStaticThreadDef_t;
 /* USER CODE BEGIN PTD */
+// Networking
+// the IPv6 multicast address, only ff02 is important, the rest is random
+// see https://en.wikipedia.org/wiki/Solicited-node_multicast_address for why ff02 matters
+#define MAX_MULTICAST_CHANNELS 16
+#define MULTICAST_CHANNEL_LENGTH 21
+const char MULTICAST_CHANNELS[MAX_MULTICAST_CHANNELS][MULTICAST_CHANNEL_LENGTH] = {
+    "ff02::c3d0:42d2:bb01", "ff02::c3d0:42d2:bb02", "ff02::c3d0:42d2:bb03",
+    "ff02::c3d0:42d2:bb04", "ff02::c3d0:42d2:bb05", "ff02::c3d0:42d2:bb06",
+    "ff02::c3d0:42d2:bb07", "ff02::c3d0:42d2:bb08", "ff02::c3d0:42d2:bb09",
+    "ff02::c3d0:42d2:bb10", "ff02::c3d0:42d2:bb11", "ff02::c3d0:42d2:bb12",
+    "ff02::c3d0:42d2:bb13", "ff02::c3d0:42d2:bb14", "ff02::c3d0:42d2:bb15",
+    "ff02::c3d0:42d2:bb16",
+};
+
+// the port robots are listening to for vision and primitives
+const short unsigned int VISION_PORT    = 42069;
+const short unsigned int PRIMITIVE_PORT = 42070;
+
+// the port the AI receives msgs from the robot
+const short unsigned int ROBOT_STATUS_PORT = 42071;
+const short unsigned int ROBOT_LOGS_PORT   = 42072;
+
+// the port to listen to for what side of the field to defend
+const unsigned DEFENDING_SIDE_PORT = 42073;
+
+// the timeout to recv a network packet
+const int NETWORK_TIMEOUT_MS = 1000;
+
+// maximum transfer unit of the network interface
+// this is an int to avoid Wconversion with lwip
+const short unsigned int MAXIMUM_TRANSFER_UNIT_BYTES = 1500;
+
```

Comment:
these should be defined in a constants file

Reply:
good catch, I thought i cleaned up all the cruft but I missed this